### ISSUE TITLE
[Bug]: Unknown bug

### ISSUE THREAD

**Issue by CoolPlayLin** [russellbanks/Komac#480]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

![image](https://github.com/russellbanks/Komac/assets/96709586/919043d6-a098-4ebf-b0ae-8362ba5cf502)
 

Full log: https://github.com/CoolPlayLin/AutoPublish/actions/runs/8000864953/job/21850993508

**Comment by rakleed** [comment]
@russellbanks I have the same problem:
```log
> Would you like to make a pull request for Nvidia.NvidiaApp.Beta 10.0.0.499? Yes
Error:
   0: <gh_user> does not have the correct permissions to execute `CreateRef`
   1: No reference was returned after creating <package>
````
I'm using the classic token.

**Comment by CoolPlayLin** [comment]
Classic token and nightly version @russellbanks

**Comment by CoolPlayLin** [comment]
I think that's a permission issue. When I checked all of the permissions options, it works again
![image](https://github.com/russellbanks/Komac/assets/96709586/7855d080-efbb-4030-ae8d-f4fa9cc28802)
@russellbanks I think Komac needs more permissions instead of only public_repo access

**Comment by jahvari** [comment]
@russellbanks I received the same error, but once I enabled all the permission for the token, it no longer appeared.

**Comment by pbek** [comment]
I ran into the same issue today with my `public_repo` token, but I don't really want to create a token with all permissions... 😬
@russellbanks, which permissions do we currently need?

**Comment by pbek** [comment]
I wasn't even able to get a working, initial config yet...

**Comment by SSLbits** [comment]
Yeah I'm not sure if github has made API changes or if there are changes to the komac workflow, but I kinda threw everything at the wall in terms of broad scopes to the classic token before I could successfully submit a PR:
all repo, workflow, write:packages, delete packages, gist, notifications, and write:discussion. That allowed komac to create a PR to microsoft/winget-pkgs.
